pkg/pipeline/engine/jenkins/mappper.go

```diff
@@ -0,0 +1,200 @@
+package jenkins
+
+import (
+	"bytes"
+	"fmt"
+	"strings"
+
+	"github.com/pkg/errors"
+	images "github.com/rancher/rancher/pkg/image"
+	"github.com/rancher/rancher/pkg/pipeline/utils"
+	mv3 "github.com/rancher/types/apis/management.cattle.io/v3"
+	"github.com/rancher/types/apis/project.cattle.io/v3"
+)
+
+func ConvertPipelineExecutionToJenkinsPipeline(execution *v3.PipelineExecution) (PipelineJob, error) {
+	if execution == nil {
+		return PipelineJob{}, errors.New("nil pipeline execution")
+	}
+
+	if err := utils.ValidPipelineConfig(execution.Spec.PipelineConfig); err != nil {
+		return PipelineJob{}, err
```

Comment:
would it make sense to return *PipelineJob instead? To make validation in the caller stack easier. Or would it introduce some extra complexity?

Reply:
It make sense